classes/local/v1p1/endpoint_test.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace enrol_oneroster\tests\local;
+
+use advanced_testcase;
+use enrol_oneroster\local\roster\rostering;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once(__DIR__ . '/../../../../config.php');
+
+class endpoint_test extends advanced_testcase {
+
+    public function test_get_classes_for_user_returns_expected_format() {
+
+        $this->resetAfterTest(true);
+
+        // Mocking the rostering class and overriding get_user_classes
+        $mock = $this->getMockBuilder(rostering::class)
+                     ->disableOriginalConstructor()
+                     ->onlyMethods(['get_user_classes'])
+                     ->getMock();
+
+        // Defining the return value for the mocked method
+        $mock->expects($this->once())
```

Comment:
So here, you have basically just defined the response, called it, and then confirmed the response equals what you said.

what you want to do instead is define whatever HTTP call might return, then you call the code you actually want to test, and then you confirm it all lines up at the end.

This allows you to actually test that the code you want to test actually handles the mock data that you set up. Happy to chat through this more in the meeting if this needs clarifying